framework/mongo/utils.py

```diff
@@ -61,15 +62,79 @@ def wrapper(cls):
     return wrapper
 
 
-def get_or_http_error(Model, pk):
-    instance = Model.load(pk)
-    if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
-    if not instance:
-        raise HTTPError(http.NOT_FOUND, data=dict(
-            message_long="No resource with that primary key could be found"
-        ))
-    else:
+def get_or_http_error(Model, pk_or_query, display_name=None):
```

Comment:
Eventually, the api util may need to be updated to be as flexible as this implementation, but this cannot replace `get_object_or_404` as is because they use different exception classes.